apps/native-component-list/src/screens/UI/MatchedGeometryEffectScreen.ios.tsx

```diff
@@ -0,0 +1,86 @@
+import { Host, HStack, Namespace, VStack, ZStack } from '@expo/ui/swift-ui';
+import {
+  padding,
+  animation,
+  Animation,
+  background,
+  cornerRadius,
+  frame,
+  matchedGeometryEffect,
+} from '@expo/ui/swift-ui/modifiers';
+import { useId, useState } from 'react';
+import { View } from 'react-native';
+
+const boxes = ['box1', 'box2', 'box3', 'box4', 'box5', 'box6'];
+const colors = ['#FF3B30', '#007AFF', '#28CD41', '#FF9500', '#AF52DE', '#FF2D92'];
+
+export default function MatchedGeometryEffectDemo() {
+  const [selectedBox, setSelectedBox] = useState<string | null>(null);
+  const namespaceId = useId();
+
+  return (
+    <View style={{ flex: 1, backgroundColor: 'white' }}>
+      <Host style={{ flex: 1 }}>
+        <Namespace id={namespaceId}>
+          <ZStack modifiers={[animation(Animation.spring(), !!selectedBox)]}>
+            {selectedBox === null ? (
+              <VStack spacing={20} modifiers={[padding({ all: 20 })]}>
+                <HStack spacing={20}>
+                  {boxes.slice(0, 3).map((boxId, index) => (
+                    //  @ts-expect-error
```

Comment:
we can remove the restriction for HStack which requires react children. or even better to add [`Rectangle`](https://developer.apple.com/documentation/swiftui/rectangle) view

Reply:
yes, will add `Rectangle` in upcoming PR. Much needed. Will add a todo note for now to replace with a `Rectangle`.